src/live-programming.rst

```diff
@@ -26,6 +26,64 @@ Interactive Debuggers
 Program Visualization
 ---------------------
 
+**Program visualization** is the use of various techniques to enhance the human understanding of computer programs from distinct aspects including the structure of the source code and run-time behavior.
+
+There are **program visualization** systems designed to aid significantly the practice of professional programming and software engineering as well as the learning of computer science by novice programmers. And it turns out that **program visualization** doesn't have to deal with toy programs, e.g., `Frappe - Visualizing Code Dependency Graphs <https://softvis.wordpress.com/2018/08/31/frappe-visualizing-code-dependency-graphs/>`_.
```

Comment:
* Don't bold "program visualization" every time you use it
* Change the structure of this sentence to "Some program visualizations are designed to aid... while others are focused on aiding..."
* I'm not sure what "it turns out that program visualization doesn't have to deal with toy programs" means? Do you mean that program visualization systems can scale to large programs? If so, and you want to mention the Frappe work, you should do it as a normal citation, not as a link to a blog post containing a video.